flatdata-py/flatdata/lib/resources.py

```diff
@@ -171,6 +188,32 @@ def __getitem__(self, item):
             ]
         return self._mem[item + SIZE_OFFSET_IN_BYTES:item + SIZE_OFFSET_IN_BYTES + 1]
 
+    def sub_str(self, index, separator = b'\0'):
+        for i in range(index, len(self)):
+            if all(a == b for (a, b) in zip(self[i:], separator)):
```

Comment:
Is `self[i:]` not a byte-string? I would find using `startswith` more readable, e.g.:

```
b'abc\0\0def'[3:].startswith(b'\0\0')
```

Or even scrap the loop and use `find`?

Reply:
It is `bytes`, not a string. How would a version of `find` that gives me an index look like?

Reply:
https://docs.python.org/3/library/stdtypes.html#bytes.find would work? Although you need to check for `-1` as a return value in case the substring was not found

```
separator_index = self.find(separator, start=index)
# or
separator_index = self[index:].find(separator)
# then
if separator_index == -1:
  return bytes(self[index]).decode("utf-8")
return bytes(self[index:separator_index).decode("utf-8")
```

Reply:
`AttributeError: 'memoryview' object has no attribute 'find'`

And converting to `bytes` reads all the data into memory

Reply:
Hmm, still find the zip solution a bit obscure, but I guess there is now easy way around it